pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/BrokerRegistry.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.broker.loadbalance.extensions;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.BiConsumer;
+import org.apache.pulsar.broker.PulsarServerException;
+import org.apache.pulsar.broker.loadbalance.extensions.data.BrokerLookupData;
+import org.apache.pulsar.metadata.api.MetadataStoreException;
+import org.apache.pulsar.metadata.api.NotificationType;
+
+/**
+ * Responsible for registering the current Broker lookup info to
+ * the distributed store (e.g. Zookeeper) for broker discovery.
+ */
+public interface BrokerRegistry extends AutoCloseable {
+
+    /**
+     * Start broker registry.
+     */
+    void start() throws PulsarServerException;
+
+    /**
+     * Register local broker to metadata store.
+     */
+    void register() throws MetadataStoreException;
+
+    /**
+     * Unregister the broker.
+     *
+     * Same as {@link org.apache.pulsar.broker.loadbalance.ModularLoadManager#disableBroker()}
+     */
+    void unregister() throws MetadataStoreException;
+
+    /**
+     * Get the current broker lookup service address.
+     *
+     * @return The service url without the protocol prefix, 'http://'. e.g. broker-xyz:port
+     */
+    String getLookupServiceAddress();
+
+    /**
+     * Get available brokers.
+     *
+     * @return The brokers service url list.
+     */
+    List<String> getAvailableBrokers();
+
+    /**
+     * Async get available brokers.
+     *
+     * @return The brokers service url list.
+     */
+    CompletableFuture<List<String>> getAvailableBrokersAsync();
```

Comment:
I think we don't need a synchronous API in the interface. The synchronous APIs are usually exposed to application users for code simplicity. But in the core code of Pulsar, we should use asynchronous APIs as much as possible.

If you want a synchronous API, you can simply use the `join` or `get` method, which is determined by if the exception should be ignored. BTW, when expose a synchronous API to users, the exception should not be ignored.

Reply:
Updated.